### PR TITLE
I implemented email registration.

### PR DESCRIPTION
**Pull request by thiago-19960626**

### CODE REVIEW

File: src/actions/register.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+var { API_URL } = require('../PLEnv');
+var { Action, ThunkAction } = require('./types');
+
+function findByUsername(username: string){
+    return new Promise((resolve, reject) => {
+        fetch(`https://api-staging.powerli.ne/api-public/users/?username=` + username, {
```
Comment (jterps08) [inline src/actions/register.js]: I do not think the "https://api-staging.powerli.ne" should be here> I think there's supposed to be a flexible reference that we can change at a master level when we are deciding which environment to create a build for... @awis0330 can confirm

File: src/actions/register.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        })
+        .catch(err => {
+            reject(err);
+        });
+    });
+}
+
+function register(data){
+    return new Promise((resolve, reject) => {
+        fetch(`https://api-staging.powerli.ne/api/secure/registration`, {
```
Comment (jterps08) [inline src/actions/register.js]: same here

File: src/components/auth/Register.js [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+            if(password.length < 6){
+                alert("Password must be six characters long");
+                return;
+            }
+            if(confirm_password == ""){
+                alert("Confirm password is empty.");
+                return;
+            }
+            if(confirm_password != password){
+                alert("Password don't match with confirm password.");
```
Comment (jterps08) [inline src/components/auth/Register.js]: "Password doesn't match. Try again."

File: src/components/auth/Register.js [hunk 4]
```diff
@@ -0,0 +1,668 @@
...
+            
+            this.setState({
+                isLoading: true
+            });
+            findByUsername(username).then(users => {
+                if(users.length > 0){
+                    this.setState({
+                        isLoading: false
+                    });
+                    alert("Username already exists.");
```
Comment (jterps08) [inline src/components/auth/Register.js]: "This username is taken"

File: src/scenes/dashboard/styles.js [hunk 5]
```diff
@@ -52,12 +52,6 @@ export default {
     color: PLColors.main,
     fontSize: (platform === 'ios') ? 18 : 15,
   },
-  iconP: {
```
Comment (jterps08) [inline src/scenes/dashboard/styles.js]: Why is this being removed?

File: native-base-theme/variables/platform.js [hunk 6]
```diff
@@ -143,7 +143,7 @@ export default {
 
   // FooterTab
   tabBarTextColor: '#8694ab',
-  tabBarTextSize: (platform === 'ios') ? 9 : 8,
```
Comment (jterps08) [inline native-base-theme/variables/platform.js]: Why is this being changed to smaller size?

File: native-base-theme/variables/platform.js [hunk 7]
```diff
@@ -281,7 +281,7 @@ export default {
   contentPadding: 10,
 
   get darkenHeader() {
-    return color(this.tabBgColor).darken(0.03).hexString();
```
Comment (jterps08) [inline native-base-theme/variables/platform.js]: Is this acceptable to @awis0330 ?

**Comment by thiago-19960626** [comment]
native-base-theme/variables/platform.js
<https://github.com/PowerlineApp/powerline-rn/pull/105#discussion_r123572203>
:

> @@ -143,7 +143,7 @@ export default {

   // FooterTab
   tabBarTextColor: '#8694ab',
-  tabBarTextSize: (platform === 'ios') ? 9 : 8,

I didn't change that, only forked and sent pull request ,

then it occurred

File: src/actions/register.js (hunk 1 above)
Comment (awis0330) [inline src/actions/register.js]: There is a string value of base url used throughout app.(i.e `API_URL` in `PLEnv.js` file).
You can import it by following code.
```
const { API_URL } = require('PLEnv'); 
```

**Pull request closed without merging**